backend/coreapp/views/scratch.py

```diff
@@ -463,9 +464,13 @@ def export(self, request: Request, pk: str) -> HttpResponse:
             zip_f.writestr("metadata.json", json.dumps(metadata, indent=4))
             zip_f.writestr("target.s", scratch.target_assembly.source_asm.data)
             zip_f.writestr("target.o", scratch.target_assembly.elf_object)
-            zip_f.writestr("code.c", scratch.source_code)
+
+            # TODO refactor #439
+            language = Language.PASCAL if scratch.compiler == "ido7.1pascal" else "C"
```

Comment:
while this says "ido7.1pascal" with different capitalization. Is this a bug?

Also, this should really be using `coreapp.compilers.from_id(scratch.compiler).language`.

Reply:
Good catch